tests/unit/ContentBlocksResolverTest.php

```diff
@@ -136,4 +136,108 @@ static function ( $parsed_block ) {
 		$this->assertEquals( 4, count( $parsed_blocks ) );
 		$this->assertEquals( $allowed, $actual_block_names );
 	}
+
+	/**
+	 * Test the wpgraphql_content_blocks_pre_resolve_blocks filter.
+	 */
+	public function test_pre_resolved_blocks_filter_returns_non_null() {
+		add_filter(
+			'wpgraphql_content_blocks_pre_resolve_blocks',
+			static function ( $blocks, $node, $args, $allowed_block_names ) {
+				return [
+					[
+						'blockName' => 'core/paragraph',
+						'attrs'     => [ 'content' => 'Test content' ],
+					],
+				];
+			},
+			10,
+			4
+		);
+
+		$post_id = self::factory()->post->create( [ 'post_content' => '' ] );
+		$post    = new Post( get_post( $post_id ) );
+
```

Comment:
Can we use the global $post_id here?

If not, rememeber to clean up at the bottom of the test